python/dgl/distributed/partition.py

```diff
@@ -11,7 +11,37 @@
 from ..random import choice as random_choice
 from ..data.utils import load_graphs, save_graphs, load_tensors, save_tensors
 from ..partition import metis_partition_assignment, partition_graph_with_halo, get_peak_mem
-from .graph_partition_book import BasicPartitionBook, RangePartitionBook
+from .graph_partition_book import BasicPartitionBook, RangePartitionBook, _str_to_tuple
+
+def _format_part_metadata(part_metadata, formator):
+    ''' Mainly for formating etypes with specified formator
+    '''
```

Comment:
Hard to understand. You should expand the description to give an example (e.g., what you are formatting to?) and why you need it. To me the `_format_part_metadata`, `_load_part_config` and `_dump_part_config` are very much like utilities for JSON. If so, why not put them under `utils/json_utils.py` and have proper documentation about their usage?

Reply:
these methods are used only in `partition.py`, so I place them here. They're for handling canonical etypes: `('_N', '_E', '_N')`.